prov/usnic/src/usdf_fabric.c

```diff
@@ -189,6 +190,42 @@ usdf_fill_addr_info(struct fi_info *fi, uint32_t addr_format,
 	return ret;		// fi_freeinfo() in caller frees all
 }
 
+void build_devname(struct usd_device_attrs *dap, char **bufp)
```

Comment:
> It quickly becomes a tangled web of sorrow. My $0.02:
> 
> 1. Keep it as a narrow match -- meaning that the strings match if strcmp() returns 0.
> 2. Keep the network-masked out bits of the IP address as 0 -- meaning that if the user submits the string "10.20.30.40/16", it would never match anything (because a /16 address would always be represented as "x.y.0.0" in libfabric).

👍  on this approach.  Networks should only match if the mask is the same length (`/16` can only match `/16`, never `/22`).  Furthermore, apply the mask before making the address comparison otherwise we're just asking our clients to replicate that logic in lots of other software when it should be centralized in ours.

The "wide match" stuff was always nonsense that we only hypothetically discussed as a way to try to simplify the command line UI of Open MPI's `mpirun` program.  I don't think it ever made much sense.